src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalController.kt

```diff
@@ -63,4 +75,65 @@ class GoalController(
       updatedGoalDto = goalResourceMapper.fromModelToDto(updateGoalRequest),
     )
   }
+
+  @PutMapping("{goalReference}/archive")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @PreAuthorize(HAS_EDIT_AUTHORITY)
+  @Transactional
+  fun <T> archiveGoal(
+    @Valid
+    @RequestBody
+    archiveGoalRequest: ArchiveGoalRequest,
+    @PathVariable @Pattern(regexp = PRISON_NUMBER_FORMAT) prisonNumber: String,
+    @PathVariable goalReference: UUID,
+  ): ResponseEntity<Any> {
+    return goalService.archiveGoal(
```

Comment:
I'm really not sure about this 🤔 , but I know this was the big win that you were after, in that you wanted to be able to see/follow the flow of the code, rather than it being hidden away in the global exception handler.

If we adopted this approach across all of our controllers we're going to add so much code 🤔
AFAIK the "best practice / recommended" (by spring evangelists) approach is to use a `@ControllerAdvice` - hence our global exception handler class. 
Given HMPPS is a Spring house I wonder what the benefits of deviating too far from standard practices might be?
Might we be in danger of violating the principal of least surprise? IE. other devs might be expecting a more standard/conventional approach of using a `@ControllerAdvice` class?

Reply:
Using Exceptions for control flow is generally considered an anti-pattern (I don't know why spring thought this was a good idea). It's an even worse offence in Kotlin as there are no checked exceptions so you lose the benefit of documenting your "bad" path in the method signature and, not that we have any, they do not play nicely with coroutines either, you must be very careful.
Appreciate this is a spring shop but to me, least surprise means I can see all the error cases in the controller and they match the swagger spec. If you use ControllerAdvise it happens in another class and is only testable via integration test instead of POJO unit tests. The code in the ControllerAdvice will be almost identical.